actors/builtin/power/power_actor.go

```diff
@@ -472,15 +471,39 @@ func (a Actor) processDeferredCronEvents(rt Runtime) error {
 	})
 
 	for _, event := range cronEvents {
-		_, _ = rt.Send(
+		_, code := rt.Send(
 			event.MinerAddr,
 			builtin.MethodsMiner.OnDeferredCronEvent,
 			vmr.CBORBytes(event.CallbackPayload),
 			abi.NewTokenAmount(0),
 		)
-		// The exit code is ignored. If a callback fails, this actor continues to invoke other callbacks
+		// If a callback fails, this actor continues to invoke other callbacks
 		// and persists state removing the failed event from the event queue. It won't be tried again.
-		// A log message would really help here, though.
+		// Failures are unexpected here but will result in removal of miner power
+		// A log message would really help here.
+		if code != exitcode.Ok {
+			rt.State().Transaction(&st, func() interface{} {
```

Comment:
Accumulate the failed miner addresses and then do just one state transaction, with a loop _inside_ it.